compiler/rustc_resolve/src/late.rs

```diff
@@ -670,7 +670,7 @@ struct DiagMetadata<'ast> {
     current_trait_object: Option<&'ast [ast::GenericBound]>,
 
     /// Given `where <T as Bar>::Baz: String`, suggest `where T: Bar<Baz = String>`.
-    current_where_predicate: Option<&'ast WherePredicate>,
+    current_where_predicate: Option<&'ast WherePredicateKind>,
```

Comment:
Why not keep a `WherePredicate`?

Reply:
I didn't find anywhere may use the attributes, so I changed it to `WherePredicateKind`.

Reply:
Indeed, it's better to keep a `WherePredicate` in case that we need the attributes in the future.